src/Triggers/AddVisualToPlayerTrigger.cs

```diff
@@ -0,0 +1,311 @@
+    }
+    #endregion
+
+    private static Dictionary<string, PlayerVisualModifier> knownModifiers;
+    public static bool TryGetModifier(string filePath, out PlayerVisualModifier modifier)
+    {
+        modifier = null;
+        if (knownModifiers?.TryGetValue(filePath, out modifier) ?? false) return true;
+        if (!Everest.Content.TryGet<AssetTypeXml>(STARTPATH + filePath, out ModAsset mod))
+        {
+            Logger.Log("CommunalHelper", "PlayerVisualModifier not found @ \""+STARTPATH + filePath + "\".");
+            return false;
+        }
+        if (LoadModifier(mod, out modifier))
+        {
+            knownModifiers[filePath] = modifier;
+            PrintPVM(filePath, modifier);
+            return true;
+        }
+        return false;
+        
+    }
+
+    private static void PrintPVM(string filePath, PlayerVisualModifier modifier)
+    {
+        Console.WriteLine("PlayerVisualModifier @ " + filePath);
+        Console.WriteLine("playerOffset: " + modifier.defaultPlayerOffset);
+        Console.Write("image: ");
+        if (modifier.image == null) Console.WriteLine("null");
+        else if (modifier.image is Sprite sprite) Console.WriteLine("Sprite with path " + sprite.Path ?? "null");
+        else Console.WriteLine("Image with texture asset path " + modifier.image.Texture.Metadata.PathVirtual);
+        Console.Write("Overrides:");
+        if (modifier.modifiersByAnim == null) Console.WriteLine(" none");
+        else
+        {
+            Console.WriteLine();
+            foreach (var kvp in modifier.modifiersByAnim)
+            {
+                Console.WriteLine("Anim: " + kvp.Key);
+                if (kvp.Value.@override != null) Console.WriteLine("  ReplaceWith: " + kvp.Value.@override);
+                if (kvp.Value.imagePlay != null) Console.WriteLine("  ImagePlays: " + kvp.Value.imagePlay);
+                Console.WriteLine("  PlayerOffset: " + kvp.Value.playerOffset?.ToString() ?? "none");
+                if (modifier.image != null) Console.WriteLine("  ImageOffset: " + kvp.Value.playerOffset?.ToString() ?? "none");
+            }
+        }
+    }
+
+    private static Vector2 getVectorFromXML(string text)
+    {
+        string[] pos = text.Split(',');
+        return new Vector2(Convert.ToInt32(pos[0]), Convert.ToInt32(pos[1]));
+    }
+
+    private static bool LoadModifier(ModAsset mod, out PlayerVisualModifier modifier)
+    {
+        modifier = new PlayerVisualModifier();
+        modifier.source = mod;
+        using (Stream inStream = mod.Stream)
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.Load(inStream);
+            XmlNode c = null;
+            foreach (XmlNode b in doc.ChildNodes)
+            {
+                if (b.Name == "PlayerVisualModifier")
+                {
+                    c = b;
+                    break;
+                }
+            }
+            if (c == null) return false;
```

Comment:
```suggestion
            XmlNode pvmNode = doc.SelectSingleNode("PlayerVisualModifier");
            if (pvmNode  == null) return false;
```
This could be simplified